### PR TITLE
[WIP] New storage

### LINKED ISSUE #693: Extract ensemble evaluations from storage
We should make a description for how to extract parameter values and the corresponding simulation responses from a single ensemble evaluation (formerly know as a case). The result should either be a description here in the issue or an isolated module in ERT that facilitates exporting data from storage.

### LINKED ISSUE #695: Extract observations from memory
Give a description for how to get the observations in memory.

### LINKED ISSUE #698: Design first iteration of data model
We should make a first iteration of the data model.

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #698
Resolves #695
Resolves #693 
### TODO: 
- Create separate issues for:
  - [x] FIELD param
  - [x] SURFACE param
  - [x] Custom KW
- Discussions
  - [x] What to do if ensemble already exists?
  - [x] How to save active realizations
  - [x] How to do logging
  - [x] Should we store models in separate file and not in \_\_init\_\_.py
  - [x] License
  - [x] API Name
- Implement before merge:
  - [x] Feature toggling
  - [x] Test extraction dump of parameters
  - [x] Test extraction dump of responses
  - [x] Add extraction dump in ES MDA
  - [x] Add extraction dump in IES
  - [x] Cleanup logging
  - [x] Store models in separate file and not in \_\_init\_\_.py
  - [x] Add LGPL license to storage
  - [x] Move libres facade fix to separate PR
  - [x] First iteration of storage_api

### CODE REVIEW

File: ert_shared/libres_facade.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def gather_gen_kw_data(self, case, key):
         """ :rtype: pandas.DataFrame """
         data = GenKwCollector.loadAllGenKwData(self._enkf_main, case, [key])
         if key in data:
-            return data[key].dropna()
+            return data[key].to_frame().dropna()
```
Comment (lars-petter-hauge) [inline ert_shared/libres_facade.py]: This part, although it probably should be added to keep consistent return data structure for all `get` functions, it does cause a bug.

https://github.com/equinor/ert/issues/720

Comment (sondreso) [reply ert_shared/libres_facade.py]: Agreed!

**Comment by jondequinor** [comment]
There are some normalization issues in the schema. The MM relationship tables like `Update` should really use a composite primary key of `ensemble_reference_id` and `ensemble_result_id`. Then you can drop the `id` column as an identity column, as well as the unique constraint. The schema then becomes a bit clearer, more logical. See https://weblogs.sqlteam.com/jeffs/2007/08/23/composite_primary_keys/

It also simplifies the API itself, as you don't have to `filter(…).first()` to get e.g. a realization (which is identified using an int and an ensemble_id (index-ensemble_id).

Minor improvement for this POC, but I think crucial in the long term.

**Comment by sondreso** [comment]
Superseded by https://github.com/equinor/ert/pull/834

**Pull request closed without merging**